core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -564,8 +608,8 @@ void parseExpr(GenericLexer lexer, ExpressionParserListener listener) throws Sql
                                     // timestamp with time zone '2005-04-02 12:00:00-07'
 
                                     // validate type
-                                    final int columnType = ColumnType.columnTypeOf(prevNode.token);
-                                    if (columnType < 0 || columnType > ColumnType.LONG256) {
+                                    final int columnType = ColumnType.tagOf(ColumnType.columnTypeOf(prevNode.token));
```

Comment:
this looks meaningless - columnTypeOf() does not return anything else other than tag itself